### PR TITLE
Delete assignment feature

### LINKED ISSUE #69: Can only assign volunteers who are only available on that date.
Right now, we can assign volunteers using any dates. We should assign volunteers who are only available.
We should also limit the assignment to 1 per date per person

### LINKED ISSUE #70: Delete Assignments
Right now, we only have the function to add assignments. Removing should be necessary as well.

deletea INDEX

### PR DESCRIPTION
**Pull request by Benson15912**
Fix #70 
Fix #69 

Add feature to delete assignments.
Example usage:
removeassign INDEX

Also add feature to only be able to assign available volunteers

### CODE REVIEW

File: src/main/java/seedu/address/model/assignment/AssignmentList.java [hunk 1]
```diff
@@ -23,6 +25,9 @@ public class AssignmentList implements Iterable<Assignment> {
      */
     public void add(Assignment toAdd) {
         requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateAssignmentException();
```
Comment (SimKianSeng) [inline src/main/java/seedu/address/model/assignment/AssignmentList.java:29]: Perhaps there can be an error message here?

**Review by SimKianSeng — APPROVED** [review]
Overall LGTM

File: src/main/java/seedu/address/model/assignment/Assignment.java [hunk 2]
```diff
@@ -44,6 +44,21 @@ public String getDetails() {
...
+     * and date.
+     * This defines a weaker notion of equality between two assignments and
+     * prevent volunteers from being assigned 2 assignments on the same day.
+     */
+    public boolean isAlreadyAssigned(Assignment otherAssignment) {
+        if (otherAssignment == this) {
+            return true;
+        }
+
+        return otherAssignment.availability.equals(availability)
```
Comment (xuwen-ho) [inline src/main/java/seedu/address/model/assignment/Assignment.java:58]: So are we assuming a person can only be assigned to one event per day?

**Review by xuwen-ho — APPROVED** [review]
Delete works as intended. Code is clear and well organised.

**Pull request merged**